File: pythonlibrary/aibrd/parsers/pdf.py
from ..models.brd import RawBRD


def parse_pdf(file_path: str) -> RawBRD:
    import fitz  # pymupdf
    doc = fitz.open(file_path)
    text = "\n\n".join(page.get_text() for page in doc)
    doc.close()
    return RawBRD(text=text, source=file_path, file_type="pdf")

File: vscode-extension/src/core/parsers/pdf.ts
import * as fs from 'fs'
import { RawBRD } from '../models/brd'

export async function parsePdf(filePath: string): Promise<RawBRD> {
  // eslint-disable-next-line @typescript-eslint/no-require-imports
  const pdfParse = require('pdf-parse')
  const buffer = fs.readFileSync(filePath)
  const data = await pdfParse(buffer)
  return {
    text: data.text,
    source: filePath,
    fileType: 'pdf'
  }
}
